src/main/common/draft/draftStoreClient.ts

```diff
@@ -1,55 +1,67 @@
-import { Draft } from 'app/models/draft'
 
-const draftStoreConfig = config.get<any>('draft-store')
+const endpointURL: string = `${config.get<any>('draft-store').url}/drafts`
 
-if (draftStoreConfig.apiVersion !== 'v2') {
-  throw new Error(`Draft store API ${draftStoreConfig.apiVersion} is not supported by the client`)
-}
+export default class DraftStoreClient<T extends DraftDocument> {
+  private serviceAuthToken: string
 
-function withAuthHeader (userId: number, other: any = {}) {
-  return Object.assign({
-    headers: {
-      'Authorization': `hmcts-id ${userId}`
-    }
-  }, other)
-}
+  constructor (serviceAuthToken: string) {
+    this.serviceAuthToken = serviceAuthToken
+  }
 
-export default class DraftStoreClient<T> {
-  private endpointURL: string
+  find (query: { [key: string]: string }, userAuthToken: string, deserializationFn: (value: any) => T): Promise<Draft<T>[]> {
+    const { type, ...qs } = query
 
-  constructor (draftType: string) {
-    if (!draftType || draftType.trim() === '') {
-      throw new Error('Draft type is required by the client')
-    }
-    this.endpointURL = `${draftStoreConfig.url}/api/${draftStoreConfig.apiVersion}/draft/${draftType}`
+    return request
+      .get(endpointURL, {
+        qs: qs,
+        headers: this.authHeaders(userAuthToken)
+      })
+      .then((response: any) => {
+        return response.data
+          .filter(draft => type ? draft.type === type : true)
+          .map(draft => {
+            const wrapper = new Draft()
+            wrapper.id = draft.id
+            wrapper.type = draft.type
+            wrapper.document = deserializationFn(draft.document)
+            wrapper.created = MomentFactory.parse(draft.created)
+            wrapper.updated = MomentFactory.parse(draft.updated)
+            return wrapper
+          })
+      })
   }
 
-  save (userId: number, draft: Draft): Promise<void> {
-    draft.lastUpdateTimestamp = MomentFactory.currentDateTime().unix()
-    return request.post(this.endpointURL, withAuthHeader(userId, {
-      body: draft
-    }))
+  save (draft: Draft<T>, userAuthToken: string): Promise<void> {
+    const options = {
+      headers: this.authHeaders(userAuthToken),
+      body: {
+        type: draft.type,
+        document: draft.document
+      }
+    }
+
+    if (!draft.id) {
+      return request.post(endpointURL, options)
+    } else {
+      return request.put(`${endpointURL}/${draft.id}`, options)
+    }
   }
 
-  retrieve (userId: number, deserializationFn: (value: any) => T): Promise<T> {
-    return request
-      .get(this.endpointURL, withAuthHeader(userId))
-      .then(draft => deserializationFn(draft))
-      .catch(err => {
-        if (err.statusCode === HttpStatus.NOT_FOUND) {
-          return undefined
-        } else {
-          throw err
-        }
-      })
+  delete (draft: Draft<T>, userAuthToken: string): Promise<void> {
+    return request.delete(`${endpointURL}/${draft.id}`, {
```

Comment:
shouldn't we make sure `id` is defined number?

Reply:
Good spot, will add this as a optimisation to prevent API calls when non saved draft is being deleted.